packages/flipper-plugin-react-native-performance/android/FlipperReactPerformancePlugin.java

```diff
@@ -0,0 +1,148 @@
+package com.phorestvoucher;
+
+import android.os.Process;
+import android.os.SystemClock;
+
+import com.facebook.flipper.core.FlipperConnection;
+import com.facebook.flipper.core.FlipperObject;
+import com.facebook.flipper.core.FlipperPlugin;
+import com.facebook.react.ReactInstanceManager;
+import com.facebook.react.bridge.ReactMarker;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class FlipperReactPerformancePlugin implements FlipperPlugin {
+
+    private FlipperConnection connection;
+    private long sessionStartTime;
+    private long nativeStartupDuration = 0;
+    private long scriptDownloadStart = 0;
+    private long scriptDownloadEnd = 0;
+    private long scriptExecutionStart = 0;
+    private long scriptExecutionEnd = 0;
+    private long contentAppeared = 0;
+    private long bundleSize;
+    private final ReactInstanceManager reactInstanceManager;
+
+    public FlipperReactPerformancePlugin(ReactInstanceManager reactInstanceManager) {
+        this.reactInstanceManager = reactInstanceManager;
+
+        sessionStartTime = System.currentTimeMillis();
+
+        measureNativeStartupTime();
+
+        measureBundleSize();
+
+        addReactMarkers();
+    }
+
+    private void measureBundleSize() {
+        String jsBundleFile = reactInstanceManager.getDevSupportManager().getDownloadedJSBundleFile();
+        File file = new File(jsBundleFile);
+        bundleSize = file.length();
+    }
+
+    private void measureNativeStartupTime() {
+        try {
+            long processStartTime = FlipperReactPerformancePlugin.getStartTime(Process.myPid());
+            nativeStartupDuration = SystemClock.elapsedRealtime() - processStartTime;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void addReactMarkers() {
+        ReactMarker.addListener(
+                (name, tag, instanceKey) -> {
+                    switch (name) {
+                        case DOWNLOAD_START:
+                            scriptDownloadStart = System.currentTimeMillis();
+                            break;
+                        case DOWNLOAD_END:
+                            scriptDownloadEnd = System.currentTimeMillis();
+                            break;
+                        case RUN_JS_BUNDLE_START:
+                            scriptExecutionStart = System.currentTimeMillis();
+                            break;
+                        case RUN_JS_BUNDLE_END:
+                            scriptExecutionEnd = System.currentTimeMillis();
+                            break;
+                        case CONTENT_APPEARED:
+                            contentAppeared = System.currentTimeMillis();
+                            sendMeasurements();
```

Comment:
The iOS implementation will continuously send the metrics as they complete, I think that's a nice UX.